File: static/CWE-89/python/vulnerable.py
# CWE-89: SQL Injection
# User-controlled request data is concatenated directly into a SQL query string.
# Vulnerable sink: cursor.execute() with a string-formatted query

import sqlite3
from flask import Flask, request

app = Flask(__name__)


@app.route("/user")
def get_user():
    user_id = request.args.get("id")
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE id = '%s'" % user_id
    cursor.execute(query)
    return str(cursor.fetchall())

File: static/CWE-89/java/vulnerable.java
// CWE-89: SQL Injection
// User-controlled input is concatenated directly into a SQL query string.
// Vulnerable sink: Statement.executeQuery(concatenatedQuery)

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

class CWE_89 {
    public ResultSet findUser(String userInput) throws Exception {
        Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/app", "app", "app");
        Statement stmt = conn.createStatement();
        String query = "SELECT * FROM users WHERE name = '" + userInput + "'";
        return stmt.executeQuery(query);
    }
}
